libkbfs/folder_block_ops.go

```diff
@@ -2140,17 +2141,25 @@ func (fbo *folderBlockOps) startSyncWriteLocked(ctx context.Context,
 
 	err = df.setBlockSyncing(file.tailPointer())
 	if err != nil {
-		return nil, nil, syncState, err
+		return nil, nil, syncState, nil, err
 	}
 	syncState.oldFileBlockPtrs = append(syncState.oldFileBlockPtrs, file.tailPointer())
+
+	// Capture the current de before we release the block lock, so
+	// other deferred writes don't slip in.
+	if de, ok := fbo.deCache[fileRef]; ok {
+		deCopy := de
```

Comment:
i don't think you need to make a copy, since `de` is already a copy (since it's a value type)